### DISCUSSION TITLE
How to show different language snippets?

### DISCUSSION THREAD

**Discussion by Felienne** [hedyorg/hedy#1564] [Website]
We are now supporting keywords in different languages but they are not yet shown in the UI. It is an interesting question how to show them, but maybe we can add a little tab to each code example where kids can switch between their native and English versions of the program, something like is common in APIs in multiple PL's?

![image](https://user-images.githubusercontent.com/1003685/146786073-091c1fe9-edfd-47c1-88a5-aeb5f63d6b7b.png)

We can also imagine that kids can set two languages in their profile: one for the UI (as we have now) and one for all demo code. That second option of course is a lot easier to build but then kids will be exposed less to the English language snippets which we want because they will ultimately have to use them going towards Python.

**Comment by Felienne** [comment]
What do you think @MarleenGilsing, @OnnoEbbens and @TiBiBa?

**Comment by MarleenGilsing** [comment]
What about: 
![image](https://user-images.githubusercontent.com/80678586/146787686-2d70e390-ffc1-43ec-ad78-48afdf9c5184.png)

A simple solution ( at least, it seems simple to me haha) that doesn't take up a lot of space, which a tab might. I would always put commands in the example codes in English by default, so the kids get exposed to them. But this way they could easliy change the language. 

I would also love it if the teachers could control this for their students. But that might be a different discussion.

**Reply by Felienne** [reply]
> ![image](https://user-images.githubusercontent.com/80678586/146787686-2d70e390-ffc1-43ec-ad78-48afdf9c5184.png)
> A simple solution ( at least, it seems simple to me haha) that doesn't take up a lot of space, which a tab might. 

I love this idea @MarleenGilsing! It is better than the tabs because indeed it does not require more space! I don't know if it is hard to build but I think our front-end king @TiBiBa will be able to fix this or our front-end queen @fRedelaar when she starts after the Christmas break.

> I would always put commands in the example codes in English by default, so the kids get exposed to them. But this way they could 
> easliy change the language.

I like that as a default setting too!
 
> I would also love it if the teachers could control this for their students. But that might be a different discussion.

Yeah technically that is a bit different but we can surely make this is a setting in the teacher's backend too. I will make two issues.

**Reply by TiBiBa** [reply]
I like it! I'm a bit struggling with how we could best implement this. Should the keyword language change on the whole website when we change it within this code example? I assume it would be distracting for students if; for example, the examples on the left bar are in English but the code examples in the adventures are in dutch. We can add this to the class customizations as well that teachers are able to enable/disable the option to switch languages.

**Reply by OnnoEbbens** [reply]
I like @MarleenGilsing idea to keep the commands in English by default and to give teachers control. For my class I think I would keep the commands in English because it gives the students a nice distinction between commands and variables/text.
